packages/rulesets/src/spectral/functions/xms-pageable-for-list-calls.ts

```diff
@@ -15,3 +21,18 @@ const xmsPageableForListCalls = (swaggerObj: any, _opts: any, paths: any) => {
 }
 
 export default xmsPageableForListCalls
+
+function matchAnyPatterns(patterns: RegExp[], path: string) {
+  return patterns.some((p) => p.test(path))
+}
+
+function verifyResourceGroupScope(path: string) {
+  // valid patterns:
+  // 1 /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachine/{vmName}
+  // 2 /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachine/{vmName}/providers/Microsoft.Billing/extensions/{extensionName}
+  const patterns = [
+    /^\/subscriptions\/{\w+}\/resourceGroups\/{\w+}\/providers\/.+/gi,
```

Comment:
is /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/ mandatory?
or is "/providers/Microsoft.Music/Configurations/config1" a valid pattern as well?

Reply:
Also, wouldnt this pattern capture patterns like "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/virtualMachine" to be valid and bypass the xms-pageable check?

Reply:
Added more tests and modified the code to address the pattern you mentioned, thanks